org.eclipse.winery.repository/src/main/java/org/eclipse/winery/repository/export/CSARExporter.java

```diff
@@ -162,26 +175,16 @@ public void writeCSAR(TOSCAComponentId entryId, OutputStream out) throws Archive
 		for (RepositoryFileReference ref : refMap.keySet()) {
 			String archivePath = refMap.get(ref);
 			CSARExporter.LOGGER.trace("Creating {}", archivePath);
-			ArchiveEntry archiveEntry = new ZipArchiveEntry(archivePath);
-			zos.putArchiveEntry(archiveEntry);
 			if (ref instanceof DummyRepositoryFileReferenceForGeneratedXSD) {
-				CSARExporter.LOGGER.trace("Special treatment for generated XSDs");
-				Document document = ((DummyRepositoryFileReferenceForGeneratedXSD) ref).getDocument();
-				DOMSource source = new DOMSource(document);
-				StreamResult result = new StreamResult(zos);
-				try {
-					transformer.transform(source, result);
-				} catch (TransformerException e) {
-					CSARExporter.LOGGER.debug("Could not serialize generated xsd", e);
-				}
+				addDummyRepositoryFileReferenceForGeneratedXSD(zos, transformer, (DummyRepositoryFileReferenceForGeneratedXSD) ref, archivePath);
 			} else {
-				try (InputStream is = Repository.INSTANCE.newInputStream(ref)) {
-					IOUtils.copy(is, zos);
-				} catch (Exception e) {
-					CSARExporter.LOGGER.error("Could not copy file content to ZIP outputstream", e);
+				if (ref.getParent() instanceof ArtifactTemplateDirectoryId) {
+					addArtifactTemplateToZipFile(zos, ref, archivePath);
+				} else {
+					addFileToZipArchive(zos, ref, archivePath);
+					zos.closeArchiveEntry();
```

Comment:
`zos.closeArchiveEntry` is called in `addArtifactTemplateToZipFile`. But it is not always necessary to close the archive entry after `addArtifactTemplateToZipFile` (e.g. misconfigured Git). Thus opening and closing archive entries was done where the information is available.